### PR TITLE
Add Possibility to Configure PSK based Encryption on UI

### PR DESCRIPTION
**Pull request by fzoske**
Hi,

This PR add the possibility to configure PSK based encryption for the communication with the Zabbix server (https://www.zabbix.com/documentation/3.0/manual/encryption/using_pre_shared_keys).

Best regards,
Fabian

### CODE REVIEW

**Comment by hudecof** [comment]
+1

**Comment by mimugmail** [comment]
I can have a look next week when I'm back.

File: net-mgmt/zabbix-proxy/src/opnsense/service/templates/OPNsense/Zabbixproxy/zabbix_proxy.conf [hunk 1]
```diff
@@ -61,4 +61,13 @@ Timeout={{ OPNsense.zabbixproxy.general.timeout }}
 {% endif %}
 FpingLocation=/usr/local/sbin/fping
 Fping6Location=/usr/local/sbin/fping6
+{% if helpers.exists('OPNsense.zabbixproxy.general.encryption') and OPNsense.zabbixproxy.general.encryption == '1' %}
+{% if helpers.exists('OPNsense.zabbixproxy.general.proxymode') and OPNsense.zabbixproxy.general.proxymode == '1' %}
+TLSAccept=psk
+{% else %}
+TLSConnect=psk
+{% endif %}
+TLSPSKFile=/usr/local/etc/zabbix3/zabbix_proxy/zabbix_proxy.psk
```
Comment (mimugmail) [inline net-mgmt/zabbix-proxy/src/opnsense/service/templates/OPNsense/Zabbixproxy/zabbix_proxy.conf]: The directory is wrong, you should use the same from targets file

File: net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml [hunk 2]
```diff
@@ -79,5 +79,17 @@
...
+            <default>0</default>
+            <Required>Y</Required>
+        </encryption>
+        <encryptionid type="TextField">
+            <default></default>
+            <Required>N</Required>
+        </encryptionid>
+        <encryptionkey type="TextField">
+            <default></default>
+            <Required>N</Required>
```
Comment (mimugmail) [inline net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml]: Can you add the correct validations to ID and PSK please? https://www.zabbix.com/documentation/3.0/manual/encryption/using_pre_shared_keys

**Comment by fzoske** [comment]
@mimugmail Fix it and renamed the config ids

One suggestion: Split the general and the encryption config to two tabs and namespaces (like the config of the Zabbix Agent plugin).
e.g.: `OPNsense.zabbixproxy.general.encryptionpsk` => `OPNsense.zabbixproxy.encryption.psk`

What do you think about this?

**Comment by mimugmail** [comment]
Hm, I have no opinion on this. I'm fine with General, hide with Advanced or a new tab.

**Comment by fzoske** [comment]
@mimugmail I put them under advanced for now

File: net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml [hunk 3]
```diff
@@ -79,5 +79,24 @@
...
+            <Required>Y</Required>
+            <advanced>true</advanced>
+        </encryption>
+        <encryptionidentity type="TextField">
+            <default></default>
+            <Required>N</Required>
+            <mask>/^.*{1,128}$/</mask>
+            <ValidationMessage>Should be a string between 1 and 128 characters.</ValidationMessage>
+            <advanced>true</advanced>
+        </encryptionid>
```
Comment (mimugmail) [inline net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml]: Wrong closing here ...

**Comment by fzoske** [comment]
fixed

**Comment by fzoske** [comment]
@fichtner You're right, the quantifier was wrong there. Fixed it.

**Comment by mimugmail** [comment]
Ok, now it's working fine. 
One thing, on my side it's not under advanced .. @fichtner does this requires a model bump?

@fzoske can you also bump the version in Makefile?

**Comment by fichtner** [comment]
yes to model version bump

File: net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml [hunk 4]
```diff
@@ -79,5 +79,24 @@
             <default>4</default>
             <Required>N</Required>
         </timeout>
+        <encryption type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+            <advanced>true</advanced>
```
Comment (fichtner) [inline net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml]: @fzoske oh, @mimugmail is right. advanced is for the form, not the model

last bit before merge... many thanks so far!

**Comment by fzoske** [comment]
Alright. Moved it to the form.... Somehow I missed that

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**